src/libs/ReportUtils.ts

```diff
@@ -2634,58 +2634,92 @@ function isUnreadWithMention(reportOrOption: OnyxEntry<Report> | OptionData): bo
- *  - is unread and the user was mentioned in one of the unread comments
- *  - is for an outstanding task waiting on the user
- *  - has an outstanding child expense that is waiting for an action from the current user (e.g. pay, approve, add bank account)
- *  - is either the system or concierge chat, the user free trial has ended and it didn't add a payment card yet
- *
- * @param option (report or optionItem)
- * @param parentReportAction (the report action the current report is a thread of)
- */
-function requiresAttentionFromCurrentUser(optionOrReport: OnyxEntry<Report> | OptionData, parentReportAction?: OnyxEntry<ReportAction>) {
+type ReasonAndReportActionThatRequiresAttention = {
+    reason: ValueOf<typeof CONST.REQUIRES_ATTENTION_REASONS>;
+    reportAction?: OnyxEntry<ReportAction>;
+};
+
+function getReasonAndReportActionThatRequiresAttention(
+    optionOrReport: OnyxEntry<Report> | OptionData,
+    parentReportAction?: OnyxEntry<ReportAction>,
+): ReasonAndReportActionThatRequiresAttention | null {
     if (!optionOrReport) {
-        return false;
+        return null;
     }
 
+    const reportActions = ReportActionsUtils.getAllReportActions(optionOrReport.reportID);
+
     if (isJoinRequestInAdminRoom(optionOrReport)) {
-        return true;
+        return {
+            reason: CONST.REQUIRES_ATTENTION_REASONS.HAS_JOIN_REQUEST,
+            reportAction: ReportActionsUtils.getActionableJoinRequestPendingReportAction(optionOrReport.reportID),
+        };
     }
 
     if (
         isArchivedRoom(optionOrReport, getReportNameValuePairs(optionOrReport?.reportID)) ||
         isArchivedRoom(getReportOrDraftReport(optionOrReport.parentReportID), getReportNameValuePairs(optionOrReport?.reportID))
     ) {
-        return false;
+        return null;
     }
 
     if (isUnreadWithMention(optionOrReport)) {
-        return true;
+        return {
+            reason: CONST.REQUIRES_ATTENTION_REASONS.IS_UNREAD_WITH_MENTION,
+        };
     }
 
     if (isWaitingForAssigneeToCompleteAction(optionOrReport, parentReportAction)) {
-        return true;
+        return {
+            reason: CONST.REQUIRES_ATTENTION_REASONS.IS_WAITING_FOR_ASSIGNEE_TO_COMPLETE_ACTION,
+            reportAction: Object.values(reportActions).find((action) => action.childType === CONST.REPORT.TYPE.TASK),
+        };
     }
 
     // Has a child report that is awaiting action (e.g. approve, pay, add bank account) from current user
     if (optionOrReport.hasOutstandingChildRequest) {
-        return true;
+        return {
+            reason: CONST.REQUIRES_ATTENTION_REASONS.HAS_CHILD_REPORT_AWAITING_ACTION,
+            reportAction: IOU.getIOUReportActionToApproveOrPay(optionOrReport, optionOrReport.reportID),
+        };
     }
 
     if (hasMissingInvoiceBankAccount(optionOrReport.reportID)) {
-        return true;
+        return {
+            reason: CONST.REQUIRES_ATTENTION_REASONS.HAS_MISSING_INVOICE_BANK_ACCOUNT,
```

Comment:
Why don't you return reportAction here?

Reply:
There's no reportAction to return here since only report is missing bank account